### PR TITLE
Add tests for download

### PR DESCRIPTION
**Pull request by inknos**
- test option resolve
- test option alldeps

### CODE REVIEW

File: dnf-behave-tests/dnf/dnf5/download.feature [hunk 1]
```diff
@@ -0,0 +1,60 @@
...
+   When I execute dnf with args "download abcde --resolve --alldeps"
+   Then the exit code is 0
+    And file "/{context.dnf.tempdir}/abcde-2.9.2-1.fc29.src.rpm" exists
+    And file "/{context.dnf.tempdir}/abcde-2.9.2-1.fc29.noarch.rpm" exists
+    And file "/{context.dnf.tempdir}/flac-1.3.2-8.fc29.x86_64.rpm" exists
+    And file "/{context.dnf.tempdir}/wget-1.19.5-5.fc29.x86_64.rpm" exists
+
+@not.with_dnf=4
+@dnf5
+Scenario: Download with alldeps options
```
Comment (jan-kolarik) [inline dnf-behave-tests/dnf/dnf5/download.feature:53]: This is the same test as the first one, probably just forgotten code block?

Comment (jan-kolarik) [reply dnf-behave-tests/dnf/dnf5/download.feature:53]: And also I thought the ``--alldeps`` is not intended to be used without the ``--resolve``.

File: dnf-behave-tests/dnf/dnf5/download.feature [hunk 2]
```diff
@@ -0,0 +1,60 @@
...
+@not.with_dnf=4
+@dnf5
+Scenario: Download with resolve option when a dependency is installed
+  Given I set working directory to "{context.dnf.tempdir}"
+    And I successfully execute dnf with args "install wget"
+   When I execute dnf with args "download abcde --resolve"
+   Then the exit code is 0
+    And file "/{context.dnf.tempdir}/abcde-2.9.2-1.fc29.src.rpm" exists
+    And file "/{context.dnf.tempdir}/abcde-2.9.2-1.fc29.noarch.rpm" exists
+    And file "/{context.dnf.tempdir}/flac-1.3.2-8.fc29.x86_64.rpm" exists
```
Comment (jan-kolarik) [inline dnf-behave-tests/dnf/dnf5/download.feature:37]: I was just thinking if not to add the explicit check that ``wget`` package does not exist, like:

``And file "/{context.dnf.tempdir}/wget-1.19.5-5.fc29.x86_64.rpm" does not exist``

**Review by jan-kolarik — CHANGES_REQUESTED** [review]

File: dnf-behave-tests/dnf/dnf5/download.feature (hunk 1 above)
Comment (inknos) [reply dnf-behave-tests/dnf/dnf5/download.feature:53]: > This is the same test as the first one, probably just forgotten code block?

my bad, I forgot to add `--alldeps`

> And also I thought the --alldeps is not intended to be used without the --resolve.

It should throw an error as we decided. It wasn't a defined behaviour at the time of writing those tests. I will change the scenario according to our decision.

File: dnf-behave-tests/dnf/dnf5/download.feature (hunk 2 above)
Comment (inknos) [reply dnf-behave-tests/dnf/dnf5/download.feature:37]: good idea, I would add it

File: dnf-behave-tests/dnf/dnf5/download.feature (hunk 1 above)
Comment (inknos) [reply dnf-behave-tests/dnf/dnf5/download.feature:55]: This is needed, for the tests too
https://github.com/rpm-software-management/dnf5/pull/185

File: dnf-behave-tests/dnf/dnf5/download.feature [hunk 3]
```diff
@@ -0,0 +1,59 @@
...
+    And file "/{context.dnf.tempdir}/wget-1.19.5-5.fc29.x86_64.rpm" exists
+
+@not.with_dnf=4
+@dnf5
+Scenario: Download with alldeps options
+  Given I set working directory to "{context.dnf.tempdir}"
+   When I execute dnf with args "download abcde --alldeps"
+   Then the exit code is 2
```
Comment (jan-kolarik) [inline dnf-behave-tests/dnf/dnf5/download.feature:58]: I would add also the check for the `stderr` here ...

**Comment by jan-kolarik** [comment]
Thanks!

**Pull request merged**